db/db_impl/db_impl_open.cc

```diff
@@ -532,68 +534,69 @@ Status DBImpl::Recover(
     // Note that prev_log_number() is no longer used, but we pay
     // attention to it in case we are recovering a database
     // produced by an older version of rocksdb.
-    std::vector<std::string> filenames;
+    std::vector<std::string> wal_dir_children;
     if (!immutable_db_options_.best_efforts_recovery) {
-      s = env_->GetChildren(immutable_db_options_.wal_dir, &filenames);
-    }
-    if (s.IsNotFound()) {
-      return Status::InvalidArgument("wal_dir not found",
-                                     immutable_db_options_.wal_dir);
-    } else if (!s.ok()) {
-      return s;
-    }
+      s = env_->GetChildren(immutable_db_options_.wal_dir, &wal_dir_children);
+      if (s.IsNotFound()) {
+        return Status::InvalidArgument("wal_dir not found",
+                                       immutable_db_options_.wal_dir);
+      } else if (!s.ok()) {
+        return s;
+      }
```

Comment:
The following code blocks does not have to be moved into the big `if (!immutable_db_options_.best_efforts_recovery)` block.